README.md

```diff
@@ -8,16 +8,207 @@ Local.
+modelSelector | JSON | Use the Cloudant Query selector syntax to associate models to existing data. NOTE: modelSelector and modelIndex are mutually exclusive; see [Selector syntax](https://docs.cloudant.com/cloudant_query.html#selector-syntax).
+
+### _rev property
+
+In a document, property `_rev` is the latest doc revision and must be provided when modifying the doc. Currently when updating/deleting a doc, the connector first retrieves that doc, then gets its `_rev`, then calls a modify api with `_rev` included. It makes three calls to the database and this would easily result in race condition.
+  
+Considering the nature of Cloudant, we plan to give user the flexibility to define `_rev` property in model and provide `_rev` by themselves instead of the connector. In such way user can get the revision and control conflict before using that revision to modify the document.
+
+# Setup Cloudant Instance
+
+For user that don't have a cloudant server to develop or test, here is some suggestions can help you quickly setup one.
+
+For development use, a docker container of Cloudant local is easy to setup and there is no request limit per second.
+
+Bluemix Cloudant will be more stable for production.
+
+- Cloudant local (docker image)
+
+  - No request limit.
+  - Please follow https://hub.docker.com/r/ibmcom/cloudant-developer/ to create your Cloudant local instance
+
+- Setup Cloudant on Bluemix
+
+  - Limit request per second by default.
+  - Choose Bluemix Cloudant if you already have a Bluemix account with a better situation than limited-days' free trial.
+
+  - Setup steps:
+
+    1. Open Bluemix console: https://console.ng.bluemix.net
+    1. Login with your account.
+    1. Click on "CATALOG" in navigation bar.
+    1. Search with keyword "cloudant" and choose the "Cloudant NOSQLDB" under "Data and Analytics".
+    1. Click on the green button "create" in the popup page to create your Cloudant database.
+    1. Go to "DASHBOARD" where you will see your new Cloudant DB icon under "Services".
+    1. Click on the icon, and it will direct you to the database page. Check "Service Credentials" on the left to see your credentials.
+    1. Check "Manage" then click on button "LAUNCH" to see your Cloudant dashboard.
 
-* Uses Cloudant Query (Lucene) to support ad-hoc searching.
-* [Loopback query](http://loopback.io/doc/en/lb3/Querying-data.html) support for: fields, limit, order, skip and where filters.
-* Query and filtering is performed on the database for optimal efficiency.
-* Use different DB instances per model definition.
-* Supports basic model discovery.
+- Create Cloudant DBaaS account
+  
+  - Limit request per second.
+  - Limited free trial.
+  - Sign up with https://cloudant.com/sign-up/ then you will see your Cloudant dashboard.
+  
+To view the Cloudant dashboard on both DBaaS and Bluemix, [sign in](https://cloudant.com/sign-in/) with your Cloudant username and password.
```

Comment:
I remember I can login with bluemix account, IIRC after you login bluemix and go to the cloudant instance dashboard, it just directs you to same domain as DBaaS